frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditTable/EditTableColumns.tsx

```diff
@@ -0,0 +1,149 @@
+import Input from '~components/Input'
+import Toggle from '~components/Toggle'
+
+import { createShortTextColumn } from '~features/admin-form/create/builder-and-design/utils/columnCreation'
+
+import { BASICFIELD_TO_DRAWER_META } from '../../../../../constants'
+
+import { EditTableInputs } from './EditTable'
+import { EditTableDropdown } from './EditTableDropdown'
+
+const TABLE_COLUMN_DROPDOWN_OPTIONS: ComboboxItem<
+  TableFieldBase['columns'][number]['columnType']
+>[] = [
+  {
+    ...pick(BASICFIELD_TO_DRAWER_META[BasicField.ShortText], 'icon', 'label'),
+    value: BasicField.ShortText,
+  },
+  {
+    ...pick(BASICFIELD_TO_DRAWER_META[BasicField.Dropdown], 'icon', 'label'),
+    value: BasicField.Dropdown,
+  },
+]
+
+interface EditTableColumnsProps {
+  isLoading: boolean
+}
+
+export const EditTableColumns = ({
+  isLoading,
+}: EditTableColumnsProps): JSX.Element => {
+  const { register, control, getValues } = useFormContext<EditTableInputs>()
+  const { errors } = useFormState<EditTableInputs>()
+  const { fields, append, remove } = useFieldArray<EditTableInputs>({
+    name: 'columns',
+  })
+
+  const requiredValidationRule = useMemo(
+    () => createBaseValidationRules({ required: true }),
+    [],
+  )
+
+  const handleAddColumn = useCallback(
+    () => append(createShortTextColumn()),
+    [append],
+  )
+
+  return (
+    <Stack
+      divider={<StackDivider borderColor="secondary.100" />}
+      spacing="2rem"
+    >
+      {fields.map((column, index) => (
+        <Stack key={column.id} spacing="1rem">
+          <FormControl
+            isRequired
+            isReadOnly={isLoading}
+            isInvalid={!!errors?.columns?.[index]?.title}
+          >
+            <Stack justify="space-between" align="center" direction="row">
+              <FormLabel>{`Column ${index + 1}`}</FormLabel>
+              {fields.length !== 1 && (
+                <IconButton
+                  variant="clear"
+                  colorScheme="danger"
+                  fontSize="1.25rem"
+                  icon={<BiTrash />}
+                  aria-label="Delete column"
+                  onClick={() => remove(index)}
+                />
+              )}
+            </Stack>
+            <Input
+              {...register(`columns.${index}.title`, requiredValidationRule)}
+            />
+            <FormErrorMessage>
+              {errors?.columns?.[index]?.title?.message}
+            </FormErrorMessage>
+          </FormControl>
+          <FormControl
+            isRequired
+            isReadOnly={isLoading}
+            id={`columns.${index}.columnType`}
+          >
+            <VisuallyHidden>
+              <FormLabel>Column type</FormLabel>
+            </VisuallyHidden>
+            <Controller
+              name={`columns.${index}.columnType`}
+              control={control}
+              rules={requiredValidationRule}
+              render={({ field }) => (
+                <SingleSelect
```

Comment:
should we make this not `isClearable`?

Reply:
opened https://github.com/opengovsg/FormSG/issues/3932 for similar issue in Rating field

Reply:
also found another bug where isClearable=true still allows clearing of input with the escape key. will be fixed alongside #3932